### PR TITLE
mutex.cpp: remove _XOPEN_SOURCE definition that causes compilation problem on netBSD 8 and is useless on Linux

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: src/mutex.cpp [hunk 1]
```diff
@@ -28,11 +28,6 @@
 
 /* projects.h and windows.h conflict - avoid this! */
```
Comment (kbevers) [inline src/mutex.cpp]: This comment doesn't seem very useful any more. `projects.h` -> `proj_internal.h` or just remove it altogether?

**Comment by kbevers** [comment]
@rouault are you happy with this as it is now? If so, I can upload RC4 tonight.

**Comment by rouault** [comment]
Yes, I'm happy as it

**Comment by kbevers** [comment]
Cool. RC4 coming shortly then!

**Comment by rouault** [comment]
you didn't merge yet to the 6.0 branch. will you do it or should I ?

**Comment by rouault** [comment]
I just did the cherry-picking to 6.0

**Comment by kbevers** [comment]
I merged it locally but I most have forgot to push it. Thanks.

**Pull request merged**